--- a/arch/arm/boot/dts/ls1021a-iot.dts
+++ b/arch/arm/boot/dts/ls1021a-iot.dts
@@ -58,6 +58,10 @@
 	tbi-handle = <&tbi1>;
 	phy-connection-type = "sgmii";
 	status = "okay";
+	fixed-link {
+             speed = <1000>;
+             full-duplex;
+	};
 };
 
 &enet2 {
@@ -71,11 +75,11 @@
 };
 
 &can1{
-	status = "omay";
+	status = "okay";
 };
 
 &can2{
-	status = "omay";
+	status = "okay";
 };
 
 &can3{
